src/server/array_func.c

```diff
@@ -908,6 +885,31 @@ create_subjob(job *parent, char *newjid, int *rc)
 		*rc = PBSE_IVALREQ;
 		return NULL;
 	}
+
+	/* save job and server within single transaction */
+	pbs_db_begin_trx(conn, 0, 0);
+
+	if (job_save_db(subj,SAVEJOB_NEW) != 0) {
+		job_purge(subj);
+		(void) pbs_db_end_trx(conn, PBS_DB_ROLLBACK);
+		*rc = PBSE_SAVE_ERR;
+		return ((job *)0);
+	}
+	/* save server here as part of the transaction */
+	if (svr_save_db(&server, SVR_SAVE_QUICK) != 0) {
+		job_purge(subj);
+		(void) pbs_db_end_trx(conn, PBS_DB_ROLLBACK);
+		*rc = PBSE_SAVE_ERR;
+		return ((job *)0);
+	}
+
+	if (pbs_db_end_trx(conn, PBS_DB_COMMIT) != 0) {
+		job_purge(subj);
+		*rc = PBSE_SAVE_ERR;
+		return ((job *)0);
+	}
+	subj->ji_newjob = 0; /* reset dontsave - job is now saved */
+
```

Comment:
We should evaluate whether we definitely need to save the subjob here. It is merely only been instantiated and not yet run, and will be saved again when the state changes to run. As such we do not need to save a subjob whose state has not yet changed to RUNNING. So can we try to optimize?

Otherwise, we will save twice - once now and once when we set to RUNNING.

Reply:
Done the suggested changes in rc 15 https://github.com/PBSPro/pbspro/pull/590/commits/bf1558fc183bcfedea1c42d6e44c38c45870d519